internals/daemon/daemon.go

```diff
@@ -240,35 +171,44 @@ func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	switch c.canAccess(r, user) {
-	case accessOK:
-		// nothing
-	case accessUnauthorized:
-		statusUnauthorized("access denied").ServeHTTP(w, r)
-		return
-	case accessForbidden:
-		statusForbidden("forbidden").ServeHTTP(w, r)
-		return
-	}
-
 	var rspf ResponseFunc
+	var access accessChecker
 	var rsp = statusMethodNotAllowed("method %q not allowed", r.Method)
 
 	switch r.Method {
 	case "GET":
 		rspf = c.GET
+		access = c.ReadAccess
 	case "PUT":
 		rspf = c.PUT
+		access = c.WriteAccess
 	case "POST":
 		rspf = c.POST
+		access = c.WriteAccess
 	case "DELETE":
 		rspf = c.DELETE
+		access = c.WriteAccess
 	}
 
-	if rspf != nil {
-		rsp = rspf(c, r, user)
+	if rspf == nil {
+		statusMethodNotAllowed("method %q not allowed", r.Method).ServeHTTP(w, r)
+		return
 	}
 
+	ucred, err := ucrednetGet(r.RemoteAddr)
+	if err != nil && err != errNoID {
+		logger.Noticef("Cannot parse UID from remote address %q: %s", r.RemoteAddr, err)
+		statusInternalError(err.Error()).ServeHTTP(w, r)
+		return
+	}
```

Comment:
Just to be consistent with snapd, can we move this block up to where it is in snapd's daemon.go? https://github.com/snapcore/snapd/blob/0a7a20db83b1d2a43319ab3e7beda18d9d97bd7d/daemon/daemon.go#L126